src/PactNet.Native/Verifier/NativePactVerifierPair.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using PactNet.Verifier;
+
+namespace PactNet.Native.Verifier
+{
+    /// <summary>
+    /// Native pact verifier configured pair type state
+    /// </summary>
+    internal class NativePactVerifierPair : IPactVerifierPair
+    {
+        private readonly IDictionary<string, string> verifierArgs;
+        private readonly IVerifierProvider verifier;
+        private readonly PactVerifierConfig config;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="NativePactVerifierPair"/> class.
+        /// </summary>
+        /// <param name="verifierArgs">Verifier args to populate</param>
+        /// <param name="verifier">Pact verifier provider</param>
+        /// <param name="config">Verifier config</param>
+        internal NativePactVerifierPair(IDictionary<string, string> verifierArgs, IVerifierProvider verifier, PactVerifierConfig config)
+        {
+            this.verifierArgs = verifierArgs;
+            this.verifier = verifier;
+            this.config = config;
+        }
+
+        /// <summary>
+        /// Set up the provider state setup URI so the service can configure states
+        /// </summary>
+        /// <param name="providerStateUri">Provider state URI</param>
+        /// <returns>Fluent builder</returns>
+        public IPactVerifierPair WithProviderStateUrl(Uri providerStateUri)
+        {
+            this.verifierArgs.Add("--state-change-url", providerStateUri.ToString());
+
+            return this;
+        }
+
+        /// <summary>
+        /// Filter the interactions to only those matching the given description and/or provider state
+        /// </summary>
+        /// <param name="description">Interaction description. All interactions are verified if this is null</param>
+        /// <param name="providerState">Provider state description. All provider states are verified if this is null</param>
+        /// <returns>Fluent builder</returns>
+        public IPactVerifierPair WithFilter(string description = null, string providerState = null)
```

Comment:
Not sold on optional arguments. I would add overloads for this.

Reply:
This is literally just moved as-is from where it was before without changing it